api/src/main/java/io/strimzi/api/kafka/model/type/AuthorizerType.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.type;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+public enum AuthorizerType {
```

Comment:
I'm wonder whether an `enum` is really the right way to model this. Will every authorizer have to have hard coded support in the CRD, or can we users wanting to configure their own custom one?

Reply:
Configuring a custom authorizer would not be so easy as it would need to be part of our Docker images. So I'm not sure it makes sense to have this really as a free String.

Reply:
> Configuring a custom authorizer would not be so easy as it would need to be part of our Docker images.

Like the external logging configuration, can't we add a volume to the container that reads a `ConfigMap` with the jar file in it? (We could to the same thing to support custom Kafka policies).

Reply:
We discussed this with Kafka Connect and ended up with no, we should not do this. (Immutability of containers etc.). So it is not so easy.

Also even if we allow the users to specify whatever class as authorizer, I think it would be better to do it as:
```
authorization:
  authorizer: CustomAuthorizer
  authorizer-class: cz.scholz.MySuperDuperAuthorizer
```